packages/caravan-wallets/src/bitbox.ts

```diff
@@ -0,0 +1,551 @@
+   * device is plugged in (`device.connect`) and unlocked
+   * (`device.unlock`).  Adds an `active` message at the `info` level
+   * when communicating with the device (`device.active`).
+   */
+  messages() {
+    const messages = super.messages();
+    messages.push({
+      state: PENDING,
+      level: INFO,
+      text: "Please plug in your BitBox.",
+      code: "device.setup",
+    });
+    messages.push({
+      state: ACTIVE,
+      level: INFO,
+      text: "Communicating with BitBox...",
+      code: "device.active",
+    });
+    return messages;
+  }
+
+  showPairingCodePopup(pairingCode: string): (() => void) | null {
+    if (this.showPairingCode) {
+      return this.showPairingCode(pairingCode);
+    }
+    const htmlContent = `
+<!doctype html>
+<html>
+  <head>
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta charset="utf-8">
+    <title>BitBox02 pairing</title>
+  </head>
+  <body>
+    <h1 style="text-align:center;">BitBox02 pairing code</h1>
+    <div id="code" style="white-space:pre;font-family:monospace,monospace;text-align:center;font-size:large;">${pairingCode}</div>
+  </body>
+</html>
+`;
+    const popup = window.open(
+      '',
+      'popupWindow',
+      'width=400,height=300',
+    );
+    if (popup) {
+      popup.document.write(htmlContent);
+      popup.document.close();
+      return () => {
+        popup.close();
+      };
+    }
+    return null;
+  }
+
+  async withDevice<T>(f: (device: PairedBitBox) => Promise<T>): Promise<T> {
+    const bitbox = await import('bitbox-api');
+
+    let hidePairingCode: (() => void) | null = null;
+    try {
+      const unpaired = await bitbox.bitbox02ConnectAuto(() => {
+        if (hidePairingCode) {
+          hidePairingCode();
+        }
+      })
+      const pairing = await unpaired.unlockAndPair()
+      const pairingCode = pairing.getPairingCode()
+      if (pairingCode) {
+        hidePairingCode = this.showPairingCodePopup(pairingCode);
+        // TODO: display pairing code to the user.
+        console.log('Pairing code:', pairingCode);
+      }
+      const pairedBitBox = await pairing.waitConfirm()
+      if (hidePairingCode) {
+        hidePairingCode();
+      }
+      try {
+        return await f(pairedBitBox)
+      } finally {
+        pairedBitBox.close()
+      }
+    } catch (err) {
+      const typedErr = bitbox.ensureError(err)
+      const isErrorUnknown = typedErr.code === 'unknown-js'
+      const errorMessage = isErrorUnknown ? typedErr.err! : typedErr.message
+      throw new Error(errorMessage);
+    } finally {
+      if (hidePairingCode) {
+        hidePairingCode();
+      }
+    }
+  }
+
+  async maybeRegisterMultisig(pairedBitBox: PairedBitBox, walletConfig: MultisigWalletConfig): Promise<{ scriptConfig: BtcScriptConfig, keypathAccount: string; }> {
+    const { scriptConfig, keypathAccount } = await convertMultisig(pairedBitBox, walletConfig);
```

Comment:
Could do something like: 
```suggestion
    const rootFingerprint = await pairedBitBox.rootFingerprint()
    this.verifyConfig(rootFingerprint, walletConfig)
    // or this.verifyConfig(pairedBitBox, walletConfig) which could do the fingerprint retrieval
    const { scriptConfig, keypathAccount } = convertMultisig(walletConfig);
```